src/commons_codec/transform/dynamodb.py

```diff
@@ -173,8 +173,24 @@ def values_to_update(self, keys: t.Dict[str, t.Dict[str, str]]) -> str:
             elif isinstance(key_value, str):
                 key_value = "'" + str(key_value).replace("'", "''") + "'"
 
-            if isinstance(key_value, dict):
-                key_value = repr(json.dumps(key_value))
+            elif isinstance(key_value, dict):
+                # TODO: Does it also need escaping of inner TEXT values, like the above?
+                key_value = "'" + json.dumps(key_value) + "'::OBJECT"
+
+            # TODO: ARRAY types do not support JSON syntax yet.
+            #       Let's relay them 1:1, which works for primitive inner types,
+            #       but complex ones need special treatment, like representing
+            #       OBJECTs in CrateDB-native syntax.
+            # FIXME: Find a way to use a custom JSONEncoder for that, in order to
+            #        fully support also nested elements of those.
+            # https://github.com/crate/commons-codec/issues/33
+            elif isinstance(key_value, list):
+                if key_value:
+                    if isinstance(key_value[0], dict):
+                        items = []
+                        for item in key_value:
+                            items.append("{" + ", ".join(f"{key}='{value}'" for key, value in item.items()) + "}")
+                        key_value = "[" + ",".join(items) + "]"
```

Comment:
> Handling it specially, [...] close to Python's `dict(foo='bar')` syntax notation.

This does not apply any special treatments to the OBJECT keys, and I did not to any further orientation flights yet. For example, what happens if the key is a reserved keyword, or contains special characters like dashes `-`?

Reply:
Depending on your spirits, we may want to defer further special handling to a subsequent iteration, and first check if the use case at hand already gets satisfied by that special transformation without any further adjustments.